meerk40t/gui/simulation.py

```diff
@@ -346,13 +386,39 @@ def on_plan_change(self, origin, plan_name, status):
             self.slider_progress.SetValue(self.max)
             self.request_refresh()
 
-    def pane_show(self):
-        self.context.setting(str, "units_name", "mm")
+    def update_fields(self):
+        step = self.progress
+        travel = self.cutcode.length_travel(stop_at=step)
+        cuts = self.cutcode.length_cut(stop_at=step)
+        travel /= MILS_IN_MM
+        cuts /= MILS_IN_MM
```

Comment:
This code is probably not device sensitive. Since the native units for a balor device isn't going to be a mil.

Reply:
The error here is, I think, mine. I converted everything over to native units during the cut planning process except speed. 

During cut planning, we scaled everything so that it would be in the native units of the device. This is an essential step since we are device agnostic prior to this and we sort of need things to be passed to the back-end in a 1:1 ratio. However, speeds are fairly universally given in `mm/s`, however that's a problem because native distances  are unrelated to physical units at that point. Speeds, consequently, violate this compact because they are still scaled given (partially) in physical rather than native units. -- What I actually need to do is scale the speed too so that you actively have access to speed in `native units / second` so that an M2Nano would get mil/second (or a different unit if you swapped your stepper drivers (for example)) and balor would get `galvos / second` which interesting is totally what the device accepts as real speeds.